### PR TITLE
gisaid/genbank file prep sequences

### PR DESCRIPTION
**Pull request by phoenixAja**
### Summary:
- **What:** add prefix to sampe ids depending on public repository name
- **Ticket:** [sc209564](https://app.shortcut.com/genepi/story/209564)
- **Env:** `<rdev link>`

### Checklist:
- [x] I merged latest `trunk`
- [x] I manually verified the change
- [x] I added labels to my PR
- [ ] I tested in multiple browsers
- [ ] I added relevant unit tests
- [ ] I have notified others of changes they need to make locally (migrations, jobs, package updates, etc)

### CODE REVIEW

File: src/backend/aspen/api/views/tests/test_sequences.py [hunk 1]
```diff
@@ -18,22 +22,90 @@
...
+    )
+    assert res.status_code == 200
+    expected_filename = get_fasta_filename("GenBank", group.name)
+    assert (
+        res.headers["Content-Disposition"]
+        == f"attachment; filename={expected_filename}"
+    )
+    file_contents = str(res.content, encoding="UTF-8")
+    assert "ATGCAAAAAA" in file_contents
+    assert file_contents.startswith(">SARS-CoV-2/human/")
```
Comment (jgadling) [inline src/backend/aspen/api/views/tests/test_sequences.py]: Please update these tests to ensure that the full sample names are correct!

File: src/backend/aspen/api/utils/fasta_streamer.py [hunk 2]
```diff
@@ -61,19 +75,43 @@ async def stream(self) -> AsyncGenerator[str, None]:
...
+        if self.pathogen_slug and self.public_repository_name:
+            # only get the prefix if we have enough information to proceed
+            prefix = (
+                sa.select(PathogenRepoConfig)
+                .join(Pathogen)
+                .join(PublicRepository)
+                .where(
+                    and_(
+                        Pathogen.slug == self.pathogen_slug,
+                        PublicRepository.name == self.public_repository_name,
```
Comment (jgadling) [inline src/backend/aspen/api/utils/fasta_streamer.py]: Just a headsup that this repository name is *case sensitive* and if we don't find a matching public repository, this endpoint doesn't return an error -- it just doesn't add prefixes to our sample names.

**Review by jgadling — CHANGES_REQUESTED** [review]

**Comment by phoenixAja** [comment]
thanks for updating the cli @jgadling

**Review by lvreynoso — APPROVED** [review]
lgtm.

**Review by jgadling — CHANGES_REQUESTED** [review]
the lstrip() is broken -- why aren't the tests catching it?

**Comment by phoenixAja** [comment]
the lstrip was working for me, i updated the tests to more specifically test that the old prefixes were stripped appropriately though.

**Review by jgadling — APPROVED** [review]

**Pull request merged**